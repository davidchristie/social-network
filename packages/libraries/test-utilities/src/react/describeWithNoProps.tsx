import React from "react";

import describeWithProps from "./describeWithProps";

export default function describeWithNoProps (
  Component: React.ComponentType
) {
  describeWithProps("with no props", Component, {});
}
