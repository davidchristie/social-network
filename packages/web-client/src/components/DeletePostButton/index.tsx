import { Alert, Button } from "design-system";
import React from "react";
import { Mutation } from "react-apollo";

import DeletePostMutation, {
  DeletePostData,
  DeletePostVariables,
} from "../../mutations/DeletePost";
import ProfileQuery from "../../queries/Profile";
import AccountQuery, { Result } from "../AccountQuery";
import ConfirmationModal from "../ConfirmationModal";

interface Props {
  postId: string;
}

interface State {
  isConfirmationModalOpen: boolean;
}

export class DeletePostButtonContent extends React.Component<Props, State> {
  public state = {
    isConfirmationModalOpen: false,
  };

  public render () {
    return (
      <AccountQuery>
        {({ data, error, loading }) => {
          if (loading || !data || !data.account) {
            return "Loading";
          }
          if (error) {
            return <Alert>{error.message}</Alert>;
          }
          const { account } = data;
          return (
            <Mutation<DeletePostData, DeletePostVariables>
              mutation={DeletePostMutation}
              refetchQueries={[
                {
                  query: ProfileQuery,
                  variables: {
                    id: account.profile.id,
                  },
                },
              ]}
              variables={{
                id: this.props.postId,
              }}
            >
              {(deletePost) => (
                <React.Fragment>
                  <Button onClick={this.openConfirmationModal}>
                    Delete
                  </Button>
                  <ConfirmationModal
                    onClose={this.closeConfirmationModal}
                    onConfirm={() => deletePost()}
                    open={this.state.isConfirmationModalOpen}
                  >
                    Are you sure you want to delete this post?
                  </ConfirmationModal>
                </React.Fragment>
              )}
            </Mutation>
          );
        }}
      </AccountQuery>
    );
  }

  private closeConfirmationModal = () => {
    this.setState({
      isConfirmationModalOpen: false,
    });
  }

  private openConfirmationModal = () => {
    this.setState({
      isConfirmationModalOpen: true,
    });
  }
}

const DeletePostButton: React.SFC<Props> = (props: Props) => (
  <AccountQuery>
    {(result: Result) => {
      return (
        <DeletePostButton {...props} {...result} />
      );
    }}
  </ AccountQuery>
);

export default DeletePostButton;
