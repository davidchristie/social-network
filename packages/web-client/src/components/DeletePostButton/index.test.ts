import describeWithNoProps from "../../testing/describeWithNoProps";
import DeletePostButton from "./index";

describe("DeletePostButton component", () => {
  describeWithNoProps(DeletePostButton);
});
