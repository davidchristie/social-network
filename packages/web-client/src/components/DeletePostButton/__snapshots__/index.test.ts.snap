// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeletePostButton component with no props matches snapshot 1`] = `
<Query
  query={
    Object {
      "definitions": Array [
        Object {
          "directives": Array [],
          "kind": "FragmentDefinition",
          "name": Object {
            "kind": "Name",
            "value": "Image",
          },
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "alias": undefined,
                "arguments": Array [],
                "directives": Array [],
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "id",
                },
                "selectionSet": undefined,
              },
              Object {
                "alias": undefined,
                "arguments": Array [],
                "directives": Array [],
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "url",
                },
                "selectionSet": undefined,
              },
            ],
          },
          "typeCondition": Object {
            "kind": "NamedType",
            "name": Object {
              "kind": "Name",
              "value": "Image",
            },
          },
        },
        Object {
          "directives": Array [],
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "Account",
          },
          "operation": "query",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "alias": undefined,
                "arguments": Array [],
                "directives": Array [],
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "account",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "email",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "id",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "name",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "profile",
                      },
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "avatar",
                            },
                            "selectionSet": Object {
                              "kind": "SelectionSet",
                              "selections": Array [
                                Object {
                                  "directives": Array [],
                                  "kind": "FragmentSpread",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "Image",
                                  },
                                },
                              ],
                            },
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "id",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "name",
                            },
                            "selectionSet": undefined,
                          },
                        ],
                      },
                    },
                  ],
                },
              },
            ],
          },
          "variableDefinitions": Array [],
        },
      ],
      "kind": "Document",
      "loc": Object {
        "end": 220,
        "start": 0,
      },
    }
  }
>
  <Component />
</Query>
`;
